addons/mrp/models/mrp_production.py

```diff
@@ -1081,6 +1088,7 @@ def action_confirm(self):
             production.workorder_ids._action_confirm()
             # run scheduler for moves forecasted to not have enough in stock
             production.move_raw_ids._trigger_scheduler()
+            production.state = 'confirmed'
```

Comment:
There is a little improvement we could do in this PR. We could take the lines 
```python
production.move_raw_ids._adjust_procure_method()
...
production.state = 'confirmed'
``` 
out of the loop to take full advantage of the batch. What do you think of that ?

Reply:
No can do for

```
self.move_raw_ids._adjust_procure_method()
(self.move_raw_ids | self.move_finished_ids)._action_confirm()
self.workorder_ids._action_confirm()
```

because if 2 MOs have workorder_ids with the same bom, then the `workorders_by_bom` in `workorder._action_confirm` will group them together and think that the first WO of one of the MOs depends on a WO from the other MO (i.e. `test_order.test_multi_button_plan` fails). I moved the others 2 lines though :)